src/NzbDrone.Core/MediaFiles/RenameEpisodeFileService.cs

```diff
@@ -110,19 +110,46 @@ private IEnumerable<RenameEpisodeFilePreview> GetPreviews(Series series, List<Ep
 
         private void RenameFiles(List<EpisodeFile> episodeFiles, Series series)
         {
-            var renamed = new List<EpisodeFile>();
+            var notRenamed = TryRenameFiles(episodeFiles, series);
+
+            while (notRenamed.Any())
+            {
+                var newNotRenamed = TryRenameFiles(notRenamed, series);
+
+                if (newNotRenamed.Count == notRenamed.Count) break;
```

Comment:
I think we need to throw/do something for this case, otherwise it's a silent failure. Also, we either need to roll back every rename or still broadcast that the rename completed, as we'd leave files out of sync in media players.